astroid/nodes/scoped_nodes/scoped_nodes.py

```diff
@@ -1123,6 +1132,9 @@ def __init__(
         self.body: list[NodeNG] = []
         """The contents of the function body."""
 
+        self.type_params: list[nodes.TypeVar] = []
```

Comment:
According to the docs we should also add `ParamSpec` and `TypeVarTuple`:
https://docs.python.org/3.12/library/ast.html#type-parameters

Or do you plan to do those in a follow up?

Reply:
No, I'll do it now. Thanks.

Reply:
Done. I marked some fixup commits that I'll squash before rebase-'n'-merge.

We may need to revisit how we want to infer some of these nodes, but we can come back to that when it comes up.